### ISSUE TITLE
Seeds for BA.2

### ISSUE THREAD

**Issue by szhan** [jeromekelleher/sc2ts-paper#265]
A couple of simple ways to get some candidate seeds for BA.2.

1. Get the intersection of the samples in the designated sequence list from `pango-designation` and the samples sequenced by the COG-UK. Then, get the samples with the earliest collection dates (before **2022-01-08**).
2. Get the earliest samples collected in South Africa (before **2021-12-01**).

**Comment by jeromekelleher** [comment]
Using COG-UK sequences joined with the pango lineage data doesn't work because the first sample is  ERR7965207 from 2022-01-03 , which happens after a retro group of BA.2s gets added:
```
2021-12-30 Add retro group 6d8d5f9adbd4d85f04633470491c6b3e n=10 {'2021-12-24': 4, '2021-12-25': 1, '2021-12-26': 1, '2021-12-28': 2, '2021-12-29': 2} {'BA.2': 8, 'BA.2.10': 2} 
```
While it looks OK generally, it's annoying that we've got the BA.2.10 creeping in here. It may not be worth worrying about though and we should just let the algorithm do it's thing with minimal intervention.

**Comment by jeromekelleher** [comment]
Do we have the experiment accessions for the resequenced early omicron sequences discussed in the Viridian paper @szhan? Just looking for "South Africa" as the region is too broad I think. Surely there's some systematic way we can track down the sequences that were actually generated??

**Comment by szhan** [comment]
Yes, Supplementary Table S7 in this Excel file https://figshare.com/articles/dataset/Supplementary_Tables_S2-9/27195315/2?file=49784541.

**Comment by szhan** [comment]
So, let's find a suitable seed among the new African samples collected before 2021-12-30? The dates for these samples should be trustworthy.

**Comment by szhan** [comment]
I can automate this and update the notebook accordingly?

**Comment by jeromekelleher** [comment]
Yes please. Can you review #275 first so I can merge?

**Comment by szhan** [comment]
Okay, now for the Omicron sublineages, we are looking among the new African samples (not only those from South Africa).

**Comment by jeromekelleher** [comment]
Searching manually among the BA.2s that show up in the max-daily-samples=1000 version I'm working with the first from the African truth-set is SRR17461930. Would be good to cross-check that this shows up in the notebook.

**Comment by szhan** [comment]
SRR17461930 isn't in the top 10, but 20th.

**Comment by jeromekelleher** [comment]
I'm restarting from the start of Omicron now using  SRR17089886,   2021-11-17   as the BA.2 seed

That came out as the first BA.2 in my quick hacks - do you agree?

**Comment by jeromekelleher** [comment]
A tricky issue here is that we don't want to add BA.2s in too early if we do think it might be a recombinant, because we have to wait for BA.1 to get properly established in the ARG (which we think may be one of the parents). If we put BA.2 sequences in too soon, then it'll be more parsimonious to just copy from one deep parent (as there's tons of mutations anyway).

This is all very difficult - I don't think we can say very much about the possible origins of these large saltational changes with these tools. The combination of the huge numbers of mutations and poor representation of the parents in the reference panel make this a bad match for our model. We can pick out recombinations very well when the parent lineages are well represented in the ARG and when we're balancing the probability of a handful of mutations vs a recombination. Massive departures from the model like Delta and Omicron are just not a good fit.

**Comment by jeromekelleher** [comment]
OK, this has worked out quite straightforwardly I think. Taking the first few COG-UK and African samples and matching against the 2021-12-01 ARG we get the following:
```
ERR7965207 2022-01-03 [{'left': 0, 'right': 22674, 'parent': 66458}, {'left': 22674, 'right': 29904, 'parent': 301233}] 86
ERR7972740 2022-01-04 [{'left': 0, 'right': 22674, 'parent': 66458}, {'left': 22674, 'right': 29904, 'parent': 301233}] 86
SRR17461953 2021-12-07 [{'left': 0, 'right': 22674, 'parent': 66458}, {'left': 22674, 'right': 29904, 'parent': 301233}] 86
SRR17461765 2021-12-09 [{'left': 0, 'right': 22674, 'parent': 66458}, {'left': 22674, 'right': 29904, 'parent': 301233}] 86
SRR17712996 2021-12-09 [{'left': 0, 'right': 22674, 'parent': 66458}, {'left': 22674, 'right': 29904, 'parent': 301233}] 84
SRR19117196 2021-12-10 [{'left': 0, 'right': 22674, 'parent': 66458}, {'left': 22674, 'right': 29904, 'parent': 301233}] 83
SRR17461792 2021-11-27 [{'left': 0, 'right': 22674, 'parent': 66458}, {'left': 22674, 'right': 29904, 'parent': 301233}] 84
```
So, the path is absolutely consistent here and I think it's reasonable to take the earliest sequence here for  2021-11-27. The left-parent node here 301233 gets added on 2021-11-23, and is clearly a solid inference and very successful node:
![Screenshot from 2024-12-21 23-41-39](https://github.com/user-attachments/assets/c5e146ab-3ea5-4f2a-bd72-b7ee4282caba)

**Comment by jeromekelleher** [comment]
Just for later reference, using this seed results (after a few weeks) in a "push" node that descends from the original recombinant with 5 reversions. These are all reversions of mutations that happen on the root of the BA.1 lineage. I don't really know what this means, but it doesn't look to me like something that would result from choosing the wrong seed and is probably more to do with having insufficient sampling (much like Delta in #226). It will be something we need to look at quite closely, as the HMM is quite clear about BA.2 being a recombinant, given the reference panel we have.

**Comment by hyanwong** [comment]
In case it's helpful to have a graphical overview, here it is, with BA.2 on the bottom left. You can see the 5 "reversions" on the right hand branch above the "BA.2?" node: they are the ones with a solid outline, and correspond to the same coloured mutations in the long line of mutations leading to BA.1 at the top right.

Interestingly this long line of mutations also contains 4 coloured mutations at the top of the line (blue, orange, yellow) which are seen in parallel (i.e. are recurrent mutations) on the LH branch above BA.2. These are not adjacent to each other on either branch, but separated by other mutations.

I wonder if it is at all plausible that the recombinant has picked different parts of the two parents somewhat at random (i.e. there isn't a single crossover point, but a semi-scattered mix of sites from the two potential parents). I don't know if there are any biological mechanisms which could cause this.

<img width="597" alt="Image" src="https://github.com/user-attachments/assets/3ca1195f-09ad-4c32-aa72-a5f61d7333cc" />

**Comment by hyanwong** [comment]
Note that according to https://github.com/jeromekelleher/sc2ts-paper/issues/319#issuecomment-2656664819, the right hand parent of the recombinant leading to BA.2 is a huge outlier in terms of mutation density. I.e. it has accumulated far more mutations (13) in a fraction of a day from its parent (node 817219 , BA.1.17.2 / ERR7606552) than we would ever expect. For what it's worth, 5 out of these 13 are reversions.
